### ISSUE TITLE
[ERR] http: Request GET <consul url>, error: rpc error: failed to get conn: rpc error: lead thread didn't get connection from=127.0.0.1:39857

### ISSUE THREAD

**Issue by sternr** [hashicorp/consul#3801]
Consul version: 0.8.5
Both client and server are running on Centos 7

Since upgrading to 0.8.5, we're seeing the following log line on the consul agent in multiple servers (actually most servers):

[ERR] http: Request GET /v1/kv/SOME-VALUE, error: rpc error: failed to get conn: rpc error: lead thread didn't get connection from=127.0.0.1:39857
 
Looking at the server logs we see no special errors

**Comment by naaaargle** [comment]
Any updates on this? I'm seeing the same issue/message only we're running 0.8.0

**Comment by slackpad** [comment]
Hi this usually means that there's a connectivity problem between the client and that server (could be a network issue, or a heavy packet loss issue sometimes related to things like running out of CPU credits on low end cloud VMs). There was an issue in https://github.com/hashicorp/consul/issues/3463, fixed in Consul 1.0.1 where we will try another server. In older versions of Consul there was a bug where it could try the same server over and over again vs. shifting to a new one. Hope that helps!

**Comment by seafoodbuffet** [comment]
Hi there,
    I don't mean to revive an old thread and will happily open a new ticket if that's the suggestion.
We're on Consul 1.0.6 on CentOS 7. I'm seeing periodic errors like this in our logs:
```
2018/02/15 09:48:16 [DEBUG] http: Request GET /v1/kv/OUR-KV-VALUE 1.404758462s) from=127.0.0.1:59856
2018/02/15 09:48:16 [ERR] consul: "KVS.List" RPC failed to server <consul server>:8300: rpc error getting client: failed to get conn: rpc error: lead thread didn't get connection
2018/02/15 09:48:16 [ERR] http: Request GET /v1/kv/OUR-KV-VALUE, error: rpc error getting client: failed to get conn: rpc error: lead thread didn't get connection from=127.0.0.1:59770
```
If I am understanding things correctly:

The first line is Consul agent's debug saying it's gotten a GET request from the API client application

The second line is Consul agent agent saying it couldn't talk to our consul server to fulfill the request

The third line is the the Consul agent's error basically telling the client that it failed

Do I have that right? 

eventually, I DO see this `[DEBUG] manager: cycled away from server "<consul server>"` and things sort of keep working.

Based on this thread, I should conclude that something beyond Consul (e.g. firewall, networking stack) is the root cause of these consul errors?
